src/Vira/Page/RegistryPage.hs

```diff
@@ -38,10 +51,88 @@ handleListing = do
   pure $ W.layout cfg crumbs $ do
     viewRepoList cfg.linkTo samples
 
+addRepoHandler :: Repo -> Eff App.AppServantStack FormResp
+addRepoHandler repo = do
+  cfg <- ask
+  App.query (St.GetRepoByNameA repo.name) >>= \case
+    Just _repo -> do
+      log Debug $ "Repository exists " <> toText repo.name
+      -- Redirect to the existing repository page
+      let existingRepoUrl :: String = show $ linkURI $ cfg.linkTo $ LinkTo.Repo repo.name
+      pure $ addHeader (toText existingRepoUrl) $ do
+        newRepoForm cfg.linkTo
+        p_ "Repository exists"
```

Comment:
Is swapping the form needed if we are redirecting the user to the existing repo page?

Reply:
good catch, fixed in 27e70538feddecd76b94d92ebefb3eb8423ca494